### ISSUE TITLE
Deadlock (?) after switching to 2.7.0

### ISSUE THREAD

**Issue by ppedziwiatr** [kriszyp/lmdb-js#203]
Hey,

my application seems to hang completely after switching to lmdb-js version `2.7.0`.

Version `2.6.9` works properly. No other changes have been made.

Do you have any idea what might be causing the issues?

**Comment by kriszyp** [comment]
There were some changes to how locks are used to wait for transactions to be flushed to disk (seems likely related to this), but I am not sure how it would actually end up in a deadlock. Do you have any suggestions for how I could reproduce this?

**Comment by kriszyp** [comment]
Ok, thanks. And I will do some of our own load testing, and see if I spot anything.

**Comment by atomictag** [comment]
I noticed the same problem. If that helps, my setup is lmdb-darwin-x64. Version 2.7.x hangs with a minimal test:

```
const rootDb = lmdb.open(...) // storage is created on file ok
await rootDb.put("something", "anything");
```  
After I reverted to 2.6.9 I noticed that running the same test on a db created with 2.7.0 would leave the promise unresolved.
Running the same test with 2.6.9 after deleting the db store would work again as usual.

**Comment by kriszyp** [comment]
I haven't been able to reproduce this. I did find potential issue with how locks were created on MacOS, and fixed in v2.7.2, if you want to give that a try and see if it happens to make any difference.

**Comment by atomictag** [comment]
I just tried 2.7.2 and all my tests pass again without any issues 🎉.
Thank you!

**Comment by ppedziwiatr** [comment]
I'm trying to test the 2.7.2 on my M1, unfortunately I'm having this issue while doing `yarn install`:
```
error /Users/piotrpedziwiatr/projects/warp-contracts-lmdb/node_modules/lmdb: Command failed.
Exit code: 127
Command: node-gyp-build-optional-packages
Arguments:
Directory: /Users/piotrpedziwiatr/projects/warp-contracts-lmdb/node_modules/lmdb
Output:
env: node\r: No such file or directory
```

Tested on Node 18.5 and 16.16.

Version 2.7.1 installs normally...

**Comment by ppedziwiatr** [comment]
hmm, `npm install` worked properly, so it's probably sth with yarn...

**Comment by kriszyp** [comment]
Ugh, those line ending errors are the worst. Should be fixed with 2.7.3, I hope.

**Comment by ppedziwiatr** [comment]
Yup, it's working great, thank you very much :-) deadlock is also fixed on my M1
